Main/Game/Game/Game.cpp

```diff
@@ -0,0 +1,334 @@
+#include "Game.h"
+#include <ctime>
+#include <algorithm>
+#include <random>
+
+Match3Game::Match3Game() : app(sf::VideoMode(1780, 960), "Match-3 Game!"), movesLeft(5), gameOver(false), gameWon(false), isReshuffling(false), click(0), isSwap(false), isMoving(false) {
+    app.setFramerateLimit(100);
+
+    t1.loadFromFile("images/background.png");
+    gameOverTexture.loadFromFile("images/game_over_background.png");
+    gameWonTexture.loadFromFile("images/game_won_background.png");
+    shuffleTexture.loadFromFile("images/NewNodes/Mod/shuffle.png");
+
+    background.setTexture(t1);
+    gameOverBackground.setTexture(gameOverTexture);
+    gameWonBackground.setTexture(gameWonTexture);
+    shuffleButton.setTexture(shuffleTexture);
+
+    offset = sf::Vector2i((app.getSize().x - GAME_SIZE * TILE_SIZE) / 2, (app.getSize().y - GAME_SIZE * TILE_SIZE) / 2);
+
+    shuffleButton.setPosition(225, 75);
+    shuffleButton.setScale(0.25f, 0.25f);
+
+    if (!font.loadFromFile("Font/BALOO2-SEMIBOLD.TTF")) {
+        // error...
+    }
+
+    movesText.setFont(font);
+    movesText.setString("Moves: " + std::to_string(movesLeft));
+    movesText.setCharacterSize(50);
+    movesText.setFillColor(sf::Color(131, 83, 83));
+    movesText.setPosition(200, 10);
+
+    taskManager.setTask(TileType::Red, 3, "images/NewNodes/Stage3/FrogP.png");
+
+    srand(time(0));
+    for (int row = 0; row < GAME_SIZE; row++) {
+        for (int col = 0; col < GAME_SIZE; col++) {
+            const unsigned int TileType = rand() % 4;
+            switch (TileType) {
+                case 0:
+                    grid[row][col] = new RedTile(row, col);
+                    break;
+                case 1:
+                    grid[row][col] = new BlueTile(row, col);
+                    break;
+                case 2:
+                    grid[row][col] = new GreenTile(row, col);
+                    break;
+                case 3:
+                    grid[row][col] = new YellowTile(row, col);
+                    break;
+            }
+        }
+    }
+}
+
+void Match3Game::run() {
```

Comment:
too many code for just one function. Better split it by the tasks each part does:
- handleSystemEvent()
- onMouseLeftPressed()
OnClick1()
OnClikc2()
calculate dxDy()
handleMovingState()
etc